Suvorov.LNU.TwitterClone.Algorithms/DetectTagsInPost.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Suvorov.LNU.TwitterClone.Algorithms
+{
+    public class DetectTagsInPost
```

Comment:
In the ExtractTagsFromPost method, the usage of Task.Run to split the userPost into words seems unnecessary and adds unnecessary overhead. Since the splitting operation is not I/O bound, it can be performed synchronously without the need for asynchronous execution.
The RemoveTagsFromPost method also uses Task.Run for splitting the userPost into words, which is redundant. Additionally, the method doesn't return the cleaned post after removing tags. It only populates a list of cleaned words but doesn't join them back together into a string. Adding a string.Join operation at the end to concatenate the cleaned words into cleanedPost would be required.

Reply:
Ok, I will try this improvement